### ISSUE TITLE
Several Python tests fail

### ISSUE THREAD

**Issue by yurivict** [NanoComp/meep#2434]
```
/usr/local/lib/python3.9/site-packages/numpy/core/getlimits.py:518: UserWarning: The value of the smallest subnormal for <class 'numpy.float64'> type is zero.
  setattr(self, word, getattr(machar, word).flat[0])
/usr/local/lib/python3.9/site-packages/numpy/core/getlimits.py:89: UserWarning: The value of the smallest subnormal for <class 'numpy.float64'> type is zero.
  return self._float_to_str(self.smallest_subnormal)
/usr/local/lib/python3.9/site-packages/numpy/core/getlimits.py:518: UserWarning: The value of the smallest subnormal for <class 'numpy.float32'> type is zero.
  setattr(self, word, getattr(machar, word).flat[0])
/usr/local/lib/python3.9/site-packages/numpy/core/getlimits.py:89: UserWarning: The value of the smallest subnormal for <class 'numpy.float32'> type is zero.
  return self._float_to_str(self.smallest_subnormal)
Traceback (most recent call last):
  File "/usr/ports/science/meep/work/meep-1.26.0/python/./tests/test_adjoint_solver.py", line 4, in <module>
    import meep.adjoint as mpa
  File "/usr/ports/science/meep/work/meep-1.26.0/python/meep/adjoint/__init__.py", line 13, in <module>
    from .optimization_problem import OptimizationProblem
  File "/usr/ports/science/meep/work/meep-1.26.0/python/meep/adjoint/optimization_problem.py", line 5, in <module>
    from autograd import grad, jacobian
  File "/usr/local/lib/python3.9/site-packages/autograd/__init__.py", line 2, in <module>
    from .differential_operators import (
  File "/usr/local/lib/python3.9/site-packages/autograd/differential_operators.py", line 13, in <module>
    import autograd.numpy as np
  File "/usr/local/lib/python3.9/site-packages/autograd/numpy/__init__.py", line 2, in <module>
    from .numpy_wrapper import *
  File "/usr/local/lib/python3.9/site-packages/autograd/numpy/numpy_wrapper.py", line 31, in <module>
    wrap_namespace(_np.__dict__, globals())
  File "/usr/local/lib/python3.9/site-packages/autograd/numpy/numpy_wrapper.py", line 20, in wrap_namespace
    int_types = {_np.int, _np.int8, _np.int16, _np.int32, _np.int64, _np.integer}
  File "/usr/local/lib/python3.9/site-packages/numpy/__init__.py", line 284, in __getattr__
    raise AttributeError("module {!r} has no attribute "
AttributeError: module 'numpy' has no attribute 'int'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/ports/science/meep/work/meep-1.26.0/python/./tests/test_adjoint_solver.py", line 6, in <module>
    import adjoint as mpa
  File "/usr/ports/science/meep/work/meep-1.26.0/python/adjoint/__init__.py", line 13, in <module>
    from .optimization_problem import OptimizationProblem
  File "/usr/ports/science/meep/work/meep-1.26.0/python/adjoint/optimization_problem.py", line 5, in <module>
    from autograd import grad, jacobian
  File "/usr/local/lib/python3.9/site-packages/autograd/__init__.py", line 2, in <module>
    from .differential_operators import (
  File "/usr/local/lib/python3.9/site-packages/autograd/differential_operators.py", line 13, in <module>
    import autograd.numpy as np
  File "/usr/local/lib/python3.9/site-packages/autograd/numpy/__init__.py", line 2, in <module>
    from .numpy_wrapper import *
  File "/usr/local/lib/python3.9/site-packages/autograd/numpy/numpy_wrapper.py", line 31, in <module>
    wrap_namespace(_np.__dict__, globals())
  File "/usr/local/lib/python3.9/site-packages/autograd/numpy/numpy_wrapper.py", line 20, in wrap_namespace
    int_types = {_np.int, _np.int8, _np.int16, _np.int32, _np.int64, _np.integer}
  File "/usr/local/lib/python3.9/site-packages/numpy/__init__.py", line 284, in __getattr__
    raise AttributeError("module {!r} has no attribute "
AttributeError: module 'numpy' has no attribute 'int'
```

meep-1.26.0
py39-numpy-1.24.1,1
Python-3.9
FreeBSD 13.1

**Comment by mrbaozi** [comment]
Your test fails here: `int_types = {_np.int, _np.int8, _np.int16, _np.int32, _np.int64, _np.integer}` - autograd is trying to use `np.int`, which was removed in numpy 1.24. However, this was fixed in HIPS/autograd@01eacff, which leads me to believe that your version of autograd is older than that.

**Comment by yurivict** [comment]
Yes, autograd was of version 1.3 which is old.

Thank you for your analysis.
